tools/smoke/internal/smoke.go

```diff
@@ -86,7 +86,7 @@ func New(logger log.Logger, cfg Config) (*Smoke, error) {
 			w.WriteHeader(http.StatusOK)
 		}
 		if cfg.SimulateErrors {
-			rand.Seed(time.Now().UnixNano())
+			rand.New(rand.NewSource(1)).Seed(time.Now().UnixNano())
```

Comment:
This doesn't seem right, you're creating a new random number generator, seeding it, but then never using it anywhere.

Reply:
Your right. The code is deprecated in 1.20, added a nolint check for it for the time being.